### ISSUE TITLE
Test failure: baseservices\\threading\\paramthreadstart\\ThreadStartUShort_1\\ThreadStartUShort_1.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#41930]
failed in job: [ runtime-coreclr ilasm 20200905.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=803706&view=ms.vss-test-web.build-test-results-tab&runId=25452210&resultId=103102&paneView=debug)

CoreCLR Windows_NT arm64 Checked ilasmroundtrip @ Windows.10.Arm64v8.Open

Error message
~~~
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'D:\h\w\B17D0988\w\B00E09B4\e\baseservices\threading\paramthreadstart\ThreadStartUShort\ThreadStartUShort.asm.dll'. The process cannot access the file because it is being used by another process. (0x80070020)
File name: 'D:\h\w\B17D0988\w\B00E09B4\e\baseservices\threading\paramthreadstart\ThreadStartUShort\ThreadStartUShort.asm.dll'

Return code:      1
Raw output file:      D:\h\w\B17D0988\w\B00E09B4\e\baseservices\threading\Reports\baseservices.threading\paramthreadstart\ThreadStartUShort_1\ThreadStartUShort_1.output.txt
Raw output:
BEGIN EXECUTION
 cmd /c ..\ThreadStartUShort\ThreadStartUShort.cmd
BEGIN EXECUTION
D:\h\w\B17D0988\p\ildasm.exe /raweh /out=ThreadStartUShort.dasm.il ThreadStartUShort.dll
D:\h\w\B17D0988\p\ilasm.exe /output=ThreadStartUShort.asm.dll  ThreadStartUShort.dasm.il

Microsoft (R) .NET IL Assembler.  Version 6.0.0-ci
Copyright (c) Microsoft Corporation.  All rights reserved.
Assembling 'ThreadStartUShort.dasm.il'  to EXE --> 'ThreadStartUShort.asm.dll'
Source file is UTF-8

Assembled method ThreadStartUShort::Main
Assembled method ThreadStartUShort::Run
Assembled method ThreadStartUShort::ThreadWorker
Assembled method ThreadStartUShort::.ctor
Creating PE file

Emitting classes:
Class 1:ThreadStartUShort

Emitting fields and methods:
Global 
Class 1Fields: 1;Methods: 4;
Resolving local member refs: 7 -> 7 defs, 0 refs, 0 unresolved

Emitting events and properties:
Global 
Class 1
Resolving local member refs: 0 -> 0 defs, 0 refs, 0 unresolved
Writing PE file
Operation completed successfully
 "D:\h\w\B17D0988\p\corerun.exe" ThreadStartUShort.asm.dll min 
END EXECUTION OF IL{D}ASM BINARY - FAILED -532462766 vs 100
Expected: 0
Actual: 1
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=D:\h\w\B17D0988\p
> D:\h\w\B17D0988\w\B00E09B4\e\baseservices\threading\paramthreadstart\ThreadStartUShort_1\ThreadStartUShort_1.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._paramthreadstart_ThreadStartUShort_1_ThreadStartUShort_1_._paramthreadstart_ThreadStartUShort_1_ThreadStartUShort_1_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.arm64.Checked\TestWrappers\baseservices.threading\baseservices.threading.XUnitWrapper.cs:line 31083
~~~

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr ilasm 20201107.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=878803&view=ms.vss-test-web.build-test-results-tab&runId=28117950&resultId=105089&paneView=debug)

failed test: baseservices\\threading\\paramthreadstart\\ThreadStartDouble\\ThreadStartDouble.cmd

CoreCLR windows arm64 Checked ilasmroundtrip @ Windows.10.Arm64v8.Open


Error message
~~~
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'D:\h\w\A90209A4\w\AA0608FD\e\baseservices\threading\paramthreadstart\ThreadStartDouble\ThreadStartDouble.asm.dll'. The process cannot access the file because it is being used by another process. (0x80070020)
File name: 'D:\h\w\A90209A4\w\AA0608FD\e\baseservices\threading\paramthreadstart\ThreadStartDouble\ThreadStartDouble.asm.dll'

Return code:      1
Raw output file:      D:\h\w\A90209A4\w\AA0608FD\e\baseservices\threading\Reports\baseservices.threading\paramthreadstart\ThreadStartDouble\ThreadStartDouble.output.txt
Raw output:
BEGIN EXECUTION
D:\h\w\A90209A4\p\ildasm.exe /raweh /out=ThreadStartDouble.dasm.il ThreadStartDouble.dll
D:\h\w\A90209A4\p\ilasm.exe /output=ThreadStartDouble.asm.dll  ThreadStartDouble.dasm.il

Microsoft (R) .NET IL Assembler.  Version 6.0.0-ci
Copyright (c) Microsoft Corporation.  All rights reserved.
Assembling 'ThreadStartDouble.dasm.il'  to EXE --> 'ThreadStartDouble.asm.dll'
Source file is UTF-8

Assembled method ThreadStartDouble::Main
Assembled method ThreadStartDouble::Run
Assembled method ThreadStartDouble::ThreadWorker
Assembled method ThreadStartDouble::.ctor
Creating PE file

Emitting classes:
Class 1:ThreadStartDouble

Emitting fields and methods:
Global 
Class 1Fields: 1;Methods: 4;
Resolving local member refs: 7 -> 7 defs, 0 refs, 0 unresolved


Emitting events and properties:
Global 
Class 1
Resolving local member refs: 0 -> 0 defs, 0 refs, 0 unresolved
Writing PE file
Operation completed successfully
 "D:\h\w\A90209A4\p\corerun.exe" ThreadStartDouble.asm.dll min 
END EXECUTION OF IL{D}ASM BINARY - FAILED -532462766 vs 100
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=D:\h\w\A90209A4\p
> D:\h\w\A90209A4\w\AA0608FD\e\baseservices\threading\paramthreadstart\ThreadStartDouble\ThreadStartDouble.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._paramthreadstart_ThreadStartDouble_ThreadStartDouble_._paramthreadstart_ThreadStartDouble_ThreadStartDouble_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\windows.arm64.Checked\TestWrappers\baseservices.threading\baseservices.threading.XUnitWrapper.cs:line 29578
~~~

**Comment by briansull** [comment]
Possibly a duplicate of  #44139

**Comment by briansull** [comment]
Fixed with #46765
